src/vs/workbench/parts/terminal/electron-browser/windowsShellHelper.ts

```diff
@@ -79,16 +84,33 @@ export class WindowsShellHelper {
 	}
 
 	/**
-	 * Returns the innermost shell executable running in the terminal
+	 * Updates innermost shell executable and innermost shell running in the terminal
```

Comment:
This comment is a little vague, something like this might be better:

> Queries the OS and gets the inner-most shell running in the terminal as well as the program the shell is running if it exists.